nft_ingester/src/backfiller.rs

```diff
@@ -0,0 +1,702 @@
+        }
+    }
+
+    /// Run the backfiller task.
+    async fn run(&mut self) {
+        loop {
+            match self.get_trees_to_backfill().await {
+                Ok(backfill_trees) => {
+                    if backfill_trees.len() == 0 {
+                        // If there are no trees to backfill, wait for a notification on the channel.
+                        let _notification = self.listener.recv().await.unwrap();
+                    } else {
+                        println!("New trees to backfill: {}", backfill_trees.len());
+
+                        // First just check if we can talk to an RPC provider.
+                        match self.rpc_client.get_version().await {
+                            Ok(version) => println!("RPC client version {version}"),
+                            Err(err) => {
+                                println!("RPC client error {err}");
+                                self.sleep_and_increase_delay().await;
+                                continue;
+                            }
+                        }
+
+                        for backfill_tree in backfill_trees {
+                            // Get the tree out of nested structs.
+                            let tree = backfill_tree.unique_tree.tree;
+                            let tree_str = bs58::encode(&tree).into_string();
+                            println!("Backfilling tree: {tree_str}");
+
+                            // Call different methods based on whether tree needs to be backfilled
+                            // completely from seq number 1 or just have any gaps in seq number
+                            // filled.
+                            let result = if backfill_tree.backfill_from_seq_1 {
+                                self.backfill_tree_from_seq_1(&tree).await
+                            } else {
+                                self.fetch_and_plug_gaps(&tree).await
+                            };
+
+                            match result {
+                                Ok(opt_max_seq) => {
+                                    if let Some(max_seq) = opt_max_seq {
+                                        // Debug.
+                                        println!("Successfully backfilled tree: {tree_str}");
+
+                                        // Only delete extra tree rows if fetching and plugging gaps worked.
+                                        if let Err(err) = self
+                                            .delete_extra_rows_and_mark_as_backfilled(
+                                                &tree, max_seq,
+                                            )
+                                            .await
+                                        {
+                                            println!("Error deleting rows and marking as backfilled: {err}");
+                                            self.sleep_and_increase_delay().await;
```

Comment:
Right now if anything fails during the backfill process, it will be retried.  Oh but one thing to clarify, is it will move onto all other trees before trying the failed tree again.  So even if it is choking on a one corrupt tree, it will move on and clear out all the good trees first before coming back.